python/ray/workflow/step_executor.py

```diff
@@ -203,10 +221,12 @@ def commit_step(store: workflow_storage.WorkflowStorage, step_id: "StepID",
     from ray.workflow.common import Workflow
     if isinstance(ret, Workflow):
         assert not ret.executed
-        tasks = [
-            _write_step_inputs(store, w.step_id, w.data)
-            for w in ret._iter_workflows_in_dag()
-        ]
+        tasks = []
+        for w in ret._iter_workflows_in_dag():
+            # If this is a reference to a workflow, do not checkpoint
```

Comment:
Does this also help the ones besides wait? For example,

s.step(w.step()), will we checkpoint input for s.step() to the actually object?  [sorry i don't quite remember this part.]

Reply:
I think this one just skips checkpointing the inputs of workflows returned by `ray.wait`, since they have already been checkpointed before.

Reply:
I see. But will it be the same for normal workflows? For example, s.step(w.step()), we actually don't need to checkpoint w.step()'s output right since it has already been checkpointed.

Reply:
Here is only about checkpointing inputs. All outputs of intermediate steps are not checkpointed here.